plugins/woocommerce-blocks/assets/js/data/checkout/push-changes.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * External dependencies
+ */
+import { debounce } from '@woocommerce/base-utils';
+import { select, dispatch } from '@wordpress/data';
+import type { AdditionalValues } from '@woocommerce/settings';
+import { ApiErrorResponse } from '@woocommerce/types';
+import { getSetting } from '@woocommerce/settings';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as CHECKOUT_STORE_KEY } from './constants';
+import { STORE_KEY as PAYMENT_STORE_KEY } from '../payment/constants';
+import { processErrorResponse } from '../utils';
+import { CheckoutPutData } from './types';
+
+// This is used to track and cache the local state of push changes.
+const localState = {
+	// True when the checkout data has been initialized.
+	isInitialized: false,
+	// True when a push is currently happening to avoid simultaneous pushes.
+	doingPush: false,
+	// Local cache of the last pushed checkoutData used for comparisons.
+	checkoutData: {
+		orderNotes: '',
+		additionalFields: {} as AdditionalValues,
+		activePaymentMethod: '',
+	},
+};
+
+const isCheckoutBlock = getSetting< boolean >( 'isCheckoutBlock', false );
```

Comment:
What property triggers push when we're not on a checkout page? Just trying to understand this part; it looks like a strange workaround and I'd like to understand why its in place.

Reply:
The defaultActivePaymentMethod is set only on the Checkout route, not Cart. So on the Cart block, initially the activePaymentMethod is empty, which triggers an update from https://github.com/woocommerce/woocommerce/blob/add/42079-additional-fields%3Dpersistent-updates/plugins/woocommerce-blocks/assets/js/data/payment/utils/set-default-payment-method.ts#L24

Reply:
I think there's been some discussion around having some way to tell if we're on the cart or checkout page client side as well, so I think this will be useful